crates/ruff_python_formatter/src/other/arguments.rs

```diff
@@ -54,13 +58,26 @@ impl FormatNodeRule<Arguments> for FormatArguments {
                     };
                 }
                 args => {
-                    joiner
-                        .entries(
-                            // We have the parentheses from the call so the item never need any
-                            args.iter()
-                                .map(|arg| (arg, arg.format().with_options(Parentheses::Preserve))),
-                        )
-                        .nodes(item.keywords.iter());
+                    // Consider e.g. `f(*args, a=b, *args2, **kwargs)`, where args (`*args`,
+                    // `*args2`) and keywords (`a=b`, `**kwargs`) are interleaved
+                    let mut args = args.iter().peekable();
+                    let mut keywords = keywords.iter().peekable();
+                    loop {
+                        let arg_start = args.peek().map_or(range.end(), Ranged::end);
+                        let keyword_start = keywords.peek().map_or(range.end(), Ranged::end);
+                        if arg_start < keyword_start {
+                            if let Some(arg) = args.next() {
```

Comment:
Isn't it guaranteed that `args.next()` must always return a value here?


```suggestion
                            if let Some(arg) = args.next().unwrap() {
```